packages/pg-protocol/src/parser.ts

```diff
@@ -119,6 +119,48 @@ export class Parser {
     }
   }
 
+  async* splitMessagesTransform(stream: NodeJS.ReadableStream) {
+    for await(const buffer of stream) {
+      this.mergeBuffer(buffer as Buffer)
+      const bufferFullLength = this.bufferOffset + this.bufferLength
+      let offset = this.bufferOffset
+      while (offset + HEADER_LENGTH <= bufferFullLength) {
+        // code is 1 byte long - it identifies the message type
+        const code = this.buffer[offset]
+        // length is 1 Uint32BE - it is the length of the message EXCLUDING the code
+        const length = this.buffer.readUInt32BE(offset + CODE_LENGTH)
+        const fullMessageLength = CODE_LENGTH + length
+        if (fullMessageLength + offset <= bufferFullLength) {
+          yield {
+            offset: offset + HEADER_LENGTH,
+            code,
+            length,
+            bytes: this.buffer
```

Comment:
This will probably lead to a bug as we pass the reference and if we modify it before the order complete this can lead to errors